mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/ParameterDslDeclarationEnricher.java

```diff
@@ -66,72 +77,96 @@ protected void onParameter(ParameterGroupDeclaration parameterGroup, ParameterDe
-                  .allowTopLevelDefinition(false)
-                  .allowsReferences(false);
-            }
-
-            @Override
-            public void visitString(StringType stringType) {
-              boolean isText = declaration.getLayoutModel() != null && declaration.getLayoutModel().isText();
-              builder.allowsInlineDefinition(dslConfiguration.allowsInlineDefinition() && (isText || isContent))
-                  .allowTopLevelDefinition(false)
-                  .allowsReferences(false);
-            }
-
-            @Override
-            public void visitArrayType(ArrayType arrayType) {
-              MetadataType genericType = arrayType.getType();
-              boolean supportsInline = supportsInlineDeclaration(arrayType, declaration.getExpressionSupport(),
-                                                                 dslConfiguration, isContent);
-              boolean isWrapped = allowsInlineAsWrappedType(genericType, typeCatalog);
-
-              builder.allowsInlineDefinition(dslConfiguration.allowsInlineDefinition() && (supportsInline || isWrapped))
-                  .allowTopLevelDefinition(dslConfiguration.allowTopLevelDefinition())
-                  .allowsReferences(dslConfiguration.allowsReferences());
-            }
-
-            @Override
-            public void visitAnyType(AnyType anyType) {
-              if (isReferableType(anyType)) {
-                builder.allowsReferences(dslConfiguration.allowsReferences())
-                    .allowTopLevelDefinition(false)
-                    .allowsInlineDefinition(false);
-              } else {
-                defaultVisit(anyType);
-              }
-            }
-
-            @Override
-            public void visitObject(ObjectType objectType) {
-              if (isMap(objectType)) {
-                builder.allowsInlineDefinition(dslConfiguration.allowsInlineDefinition());
-
-              } else if (!declaration.getModelProperty(InfrastructureParameterModelProperty.class).isPresent()) {
-
-                boolean supportsInline = supportsInlineDeclaration(objectType, declaration.getExpressionSupport(),
-                                                                   dslConfiguration, isContent);
-
-                boolean isWrapped = allowsInlineAsWrappedType(objectType, typeCatalog);
-
-                builder.allowsInlineDefinition(dslConfiguration.allowsInlineDefinition() &&
-                    (supportsInline || isWrapped));
-              }
-
-              builder.allowTopLevelDefinition(dslConfiguration.allowTopLevelDefinition())
-                  .allowsReferences(dslConfiguration.allowsReferences());
-            }
-
-            @Override
-            public void visitUnion(UnionType unionType) {
-              unionType.getTypes().forEach(type -> type.accept(this));
-            }
-
-          });
+          getTypeId(declaration.getType())
```

Comment:
add javadocs explaining what you're doing or split into several methods with descriptive names